### PR TITLE
Have Schema's internal list always immutable.

### PR DESCRIPTION
**Pull request by dmikurube**

### CODE REVIEW

**Comment by dmikurube** [comment]
Not sure this change is correct, but I was concerned that the contents of the internal list `List<Column> columns` can be modified from outside.

**Comment by dmikurube** [comment]
@muga Thanks for your comments! Updated the PR.

File: embulk-core/src/main/java/org/embulk/spi/Schema.java [hunk 1]
```diff
@@ -31,14 +31,25 @@ public static Builder builder()
...
     @JsonCreator
     public Schema(List<Column> columns)
     {
-        this.columns = columns;
+        if (columns instanceof ImmutableList) {
+            @SuppressWarnings("unchecked")
+            ImmutableList<Column> columnsCasted = (ImmutableList)columns;
+            this.columns = columnsCasted;
+        } else {
+            this.columns = ImmutableList.copyOf(columns);
```
Comment (dmikurube) [inline embulk-core/src/main/java/org/embulk/spi/Schema.java]: Done.

**Comment by muga** [comment]
@dmikurube Merged. thank you.

**Pull request merged**